client/extensions/woocommerce/app/settings/payments/payment-method-stripe.js

```diff
@@ -69,126 +107,265 @@ class PaymentMethodStripe extends Component {
 		this.props.onEditField( e.target.name, e.target.value );
 	}
 
-	onToggleTestMode = ( mode ) => {
-		const testmode = mode === 'test' ? 'yes' : 'no';
-		// return curried function
-		return () => {
-			this.props.onEditField( 'testmode', testmode );
-		};
+	hasNonTrivialStringValue = ( prop ) => {
+		if ( ! ( 'value' in prop ) ) {
+			return false;
+		}
+		if ( ! isString( prop.value ) ) {
+			return false;
+		}
+		return ( ! isEmpty( prop.value.trim() ) );
+	}
+
+	hasKeys = ( props ) => {
+		if ( ! props ) {
+			props = this.props;
+		}
+
+		const apiKeyList = [ 'publishable_key', 'secret_key', 'test_publishable_key', 'test_secret_key' ];
+		if ( some( pick( this.props.method.settings, apiKeyList ), prop => this.hasNonTrivialStringValue( prop ) ) ) {
+			return true;
+		}
+
+		return false;
 	}
 
-	renderEditKeyTextbox = ( setting, label, placeholder ) => {
-		const isError = this.state.missingFieldsNotice && ! setting.value.trim();
+	////////////////////////////////////////////////////////////////////////////
+	// Heading
+
+	renderHeading = () => {
+		const { stripeConnectAccount, translate } = this.props;
+		const { connectedUserID } = stripeConnectAccount;
+
+		// If we are not connected AND had no keys at mount display
+		// Take credit card payments with Stripe
+		if ( ! connectedUserID && ! this.state.hadKeysAtStart ) {
+			return (
+				<div className="payments__method-edit-header">
+					{ translate( 'Take credit card payments with Stripe' ) }
+				</div>
+			);
+		}
 
+		// Otherwise, display Manage Stripe
 		return (
-			<div>
-				<FormFieldset className="payments__method-edit-field-container">
-					<FormLabel required={ true }>
-						{ label }
-					</FormLabel>
-					<FormTextInput
-						name={ setting.id }
-						onChange={ this.onEditFieldHandler }
-						isError={ isError }
-						value={ setting.value }
-						placeholder={ placeholder } />
-					{ isError && <FormInputValidation isError text="This field is required." /> }
-				</FormFieldset>
+			<div className="payments__method-edit-header">
+				{ translate( 'Manage Stripe' ) }
 			</div>
 		);
 	}
 
-	renderKeyFields = ( isLiveMode ) => {
-		const { method, translate } = this.props;
-		const secretLabel = isLiveMode ? translate( 'Live Secret Key' ) : translate( 'Test Secret Key' );
-		const publishableLabel = isLiveMode ? translate( 'Live Publishable Key' ) : translate( 'Test Publishable Key' );
+	////////////////////////////////////////////////////////////////////////////
+	// Setup prompting
+
+	onSelectCreate = () => {
+		this.setState( { createSelected: true } );
+	}
+
+	onSelectConnect = () => {
+		this.setState( { createSelected: false } );
+	}
+
+	possiblyRenderSetupPrompt = () => {
```

Comment:
Looks like this function is doing 3 things, rendering a legacy prompt, checking to see if it should even be rendering, and finally rendering the prompt. Maybe split it up?

Reply:
Sure - will do

Reply:
Simplified a little as part of the https://github.com/Automattic/wp-calypso/commit/c83f9eebe6fa7e4e74893bf010bb49c94b136481 refactor